internal/schema-registry/command_dek_undelete.go

```diff
@@ -0,0 +1,80 @@
+package schemaregistry
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/config"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/resource"
+)
+
+func (c *command) newDekUndeleteCommand(cfg *config.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "undelete",
```

Comment:
undelete is to revert a soft-deleted DEK / KEK
cc @rayokota

Reply:
Yes, soft-deleted DEKs can be recovered.  A soft deleted DEK can be used by a consumer but not a producer.  After recovering/undeleting it, it can be used by the producer again